File: java/0941.0_Valid_Mountain_Array.java
/*
1ms 击败 100.00%使用 Java 的用户
42.41MB 击败 54.16%使用 Java 的用户
*/
class Solution {
    public boolean validMountainArray(int[] arr) {
        int start = 0, end = arr.length - 1;
        while (start < end && arr[start + 1] > arr[start]) {
            start++;
        }
        while (start < end && arr[end - 1] > arr[end]) {
            end--;
        }
        return start == end && start > 0 && end < arr.length - 1;
    }
}

File: js/0941.0_Valid_Mountain_Array.js
/*
76ms 击败 22.22%使用 JavaScript 的用户
42.34MB 击败 64.05%使用 JavaScript 的用户
*/
/**
 * @param {number[]} arr
 * @return {boolean}
 */
var validMountainArray = function(arr) {
    const n = arr.length;
    let i = 0, j = n - 1;
    while (i < j && arr[i] < arr[i + 1]) {
        i++;
    }
    while (i < j && arr[j] < arr[j - 1]) {
        j--;
    }
    return i == j && i > 0 && j < n - 1;
};
